### ISSUE TITLE
Split OutlierDetectionStep into multiple steps

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8634]
_Issue [JP-3682](https://jira.stsci.edu/browse/JP-3682) was created on JIRA by [Brett Graham](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bgraham):_

The outlier detection step currently handles several different "modes" in sometimes very different ways:
 * imaging
 * spec
 * coron
 * tso
 * ifu

The docs mention that the parameters for the step are different depending on the mode:

<https://jwst-pipeline.readthedocs.io/en/latest/jwst/outlier_detection/arguments.html>

The step spec includes many items that are entirely unused in some modes (which is not fully documented).

These modes also expect very different input types and produce different results.

Different modes use different file suffixes, none of which are currently handled by the shared suffix machinery.

It seems reasonable that these should be split into different steps. However this will require:
 * new `pars-` references for the new steps
 * replacing `pars-outlierdetectionstep`

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3682?focusedCommentId=714733#comment-714733):_

I started looking into what is needed here on the cal side.  It looks like something we'd definitely like to do eventually, and I made a quick draft of what it might look like, which can be found in [this draft pull request|[https://github.com/spacetelescope/jwst/pull/8722].]  However I sort of "jumped the gun" as this needs discussion with INS teams and would require parameter reference file changes, so assigning it to David again for input.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-3682?focusedCommentId=716903#comment-716903):_

Initial discussion at the Sep 4 JP meeting; will be considered as part of possible 11.2 work.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-3682?focusedCommentId=743053#comment-743053):_

Now that <https://jira.stsci.edu/browse/JP-3347> is resolving without the need to use the IFU algorithm on FS/MOS data (which would have confused this effort significantly) I think we can unblock this.  I know at the same time work has been going on to reorganize the documentation, per <https://jira.stsci.edu/browse/JP-3342>

[Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) , pushing this back over to you.  Is this still desirable from your side based on the recent doc updates?

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3682?focusedCommentId=938756#comment-938756):_

IMO this is still worth doing, although the documentation updates make it less urgent/crucial.  I'm going to resurrect my (now ancient) PR for this and see if the changes do indeed look helpful to folks.

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3682?focusedCommentId=940030#comment-940030):_

[David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) the changes to the JWST repository that are needed for this change are complete (pending PR reviews): #8722 

What should I do in order to get parameter reference files split by mode and changed so that they can be called from the new steps?  I'm unfamiliar with the process by which that would get done.

Separately, I have a slight concern (referencing a conversation I had with [Nadia Dencheva](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dencheva) ) that this change has not been fully considered by many folks outside the pipeline team, and while it does make the code easier to maintain and more streamlined on our end, the API changes might not necessarily be desired.  Do you have any thoughts on this?

**Comment by stscijgbot-jp** [comment]
_Comment by [Melanie Clarke](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mclarke) on [JIRA](https://jira.stsci.edu/browse/JP-3682?focusedCommentId=941554#comment-941554):_

I'm setting this on hold while we wait for INS input on whether/how to proceed.
